RELEASES.md

```diff
@@ -0,0 +1,414 @@
+# Version 2.0 Syntax (2019-11-11)
+
+The ink! version 2.0 syntax has one major philosophy:
+
+> Just. Be. Rust.
+
+To accomplish this, we take advantage of all the standard Rust types and structures and use
+[attribute macros](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) to
+tag these standard structures to be different parts of the ink! language.
+
+Anything that is not tagged with an `#[ink(...)]` attribute tag is just standard Rust, and can be
+used in and out of your contract just like standard Rust could be used!
+
+**Every valid ink! contract is required to have at least one `#[ink(constructor)]`, at least one
+`#[ink(message)]` and exactly one `#[ink(storage)]` attribute.**
+
+Follow the instructions below to understand how to migrate your ink! 1.0 contracts to this new ink!
+2.0 syntax.
+
+## Declaring a Contract
+
+The fundamental change with the new ink! syntax is how we declare a new contract.
+
+We used to wrap the whole ink! contract into a `contract!` macro. At that point, all syntax within
+the macro could be custom, and in our first iteration of the language, we used that in ways that
+made our code not really Rust anymore.
+
+Now we wrap the whole contract in a standard Rust module, and include an attribute tag to identify
+this object as part of the ink! language. This means that all of our code from this point forward
+will be valid Rust!
+
+<table style="width: 100%;">
+<tr>
+<th>Before</th>
+<th>After</th>
+</tr>
+<tr>
+<td>
+
+```rust
+contract! {
+    ...
+}
+```
+
+</td>
+<td>
+
+```rust
+#[ink::contract(version = "0.1.0")]
+mod erc20 {
+    ...
+}
+```
+
+</td>
+</tr>
+</table>
+
+> Note: we now require a mandatory ink! version in the header. You're welcome.
+
+## ink! Contract Tag
+
+The ink! contract tag can be extended to provide other configuration information about your
+contract.
+
+### Defining Custom Types
+
+We used to define types using a special `#![env = DefaultSrmlTypes]` tag.
+
+Now we simply include the type definition in the `#[ink::contract(...)]` tag:
+
+```rust
+#[ink::contract(version = "0.1.0", types = MyCustomTypes)]
+```
+
+By default, we use `DefaultSrmlTypes`, so you don't need to define anything unless you plan to use
+custom types.
+
+### Dynamic Allocation
+
+It is possible to enable the dynamic environment that allows for dynamic allocations by specifying
+`dynamic_allocations = true` in the parameters of the ink! header. This is disabled by default.
```

Comment:
We should make it clear that dynamic environment is still under research and not yet stable. Will possibly change in the future.